crates/solvers/src/domain/solver/dex/mod.rs

```diff
@@ -86,42 +101,65 @@ impl Dex {
             .filter_map(future::ready)
     }
 
+    async fn try_solve(
+        &self,
+        order: &Order,
+        dex_order: &dex::Order,
+        tokens: &auction::Tokens,
+        gas_price: auction::GasPrice,
+    ) -> Option<dex::Swap> {
+        self.rate_limiter
+            .execute_with_back_off(
+                async {
```

Comment:
nit: To avoid unnecessary indentation you can first create the future and then pass it to `execute_with_back_off()`.

Reply:
Extracted